network/wsNetwork.go

```diff
@@ -1937,6 +1937,11 @@ func ParseHostOrURL(addr string) (*url.URL, error) {
 	// This turns "[::]:4601" into "http://[::]:4601" which url.Parse can do
 	parsed, e2 := url.Parse("http://" + addr)
 	if e2 == nil {
+		// https://datatracker.ietf.org/doc/html/rfc1123#section-2
+		// first character is relaxed to allow either a letter or a digit
+		if parsed.Host[0] == ':' && (len(parsed.Host) < 2 || parsed.Host[1] != ':') {
+			return nil, errors.New("host name starts with colon")
```

Comment:
please define this as a global public error object and use it here.

Reply:
done